sound/soc/sof/intel/hda-dai.c

```diff
@@ -223,8 +226,10 @@ static int hda_link_hw_free(struct snd_pcm_substream *substream,
 		if (!link)
 			return -EINVAL;
 
-		stream_tag = hdac_stream(link_dev)->stream_tag;
-		snd_hdac_ext_link_clear_stream_id(link, stream_tag);
+		if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
+			stream_tag = hdac_stream(link_dev)->stream_tag;
+			snd_hdac_ext_link_clear_stream_id(link, stream_tag);
+		}
```

Comment:
at no point does the patch set the LOSDIV register, this is hidden behind a helper, so you have to describe more clearly what this helper does, why it's only valid on startup and why this fix cannot be handled in the helper itself?

Reply:
yes, I will refine my commit msg.

The describe of this function is:
/* snd_hdac_ext_link_set_stream_id - maps stream id to link output */